### PR TITLE
Make sourcemap generation a bit faster, roll #2

### PR DESCRIPTION
**Pull request by nicks**
Makes sourcemap generation a bit faster. Re-try of https://github.com/less/less.js/pull/2834

Also adds tests for different source map generation options.

### CODE REVIEW

File: lib/less/source-map-output.js [hunk 1]
```diff
@@ -104,12 +115,9 @@ module.exports = function (environment) {
         this._sourceMapGenerator = new this._sourceMapGeneratorConstructor({ file: this._outputFilename, sourceRoot: null });
 
         if (this._outputSourceFiles) {
-            for (var filename in this._contentsMap) {
-                if (this._contentsMap.hasOwnProperty(filename)) {
-                    var source = this._contentsMap[filename];
-                    if (this._contentsIgnoredCharsMap[filename]) {
-                        source = source.slice(this._contentsIgnoredCharsMap[filename]);
-                    }
+            for (var filename in this._contentsInfoMap) {
```
Comment (nicks) [inline lib/less/source-map-output.js]: the bug in the last PR was that this line was still referencing this._contentsMap

**Comment by nicks** [comment]
ya, it works for me locally...i don't know enough about your CI setup to know why it would be different there.

**Comment by matthew-dean** [comment]
@nicks Can you take another whack at running tests locally? There's no reason they should pass on another system. Are you running `grunt test`?

**Comment by nicks** [comment]
Ya. I think the problem is that the source-map library generates different source-maps on OSX vs Linux, and this PR is the first thing in this repo that adds tests for source-maps.

I added a replacer so that it's not checking the exact source-map contents, just that it exists.

**Comment by nicks** [comment]
(the travis test failures are spurious, appears to be a problem with your test setup)

**Pull request closed without merging**